libraries/creds.rb

```diff
@@ -1,11 +1,21 @@
 # frozen_string_literal: true
 
+require 'json'
+
 def aws_creds(key) # rubocop:disable Metrics/AbcSize
   begin
     node.run_state['letsencryptaws_creds'] ||= \
-      data_bag_item(node['letsencryptaws']['data_bag'], node['letsencryptaws']['data_bag_item']).to_hash
+      if File.exist?(temp_credentials_path)
+        JSON.parse(IO.read(temp_credentials_path))
+      else
+        data_bag_item(node['letsencryptaws']['data_bag'], node['letsencryptaws']['data_bag_item']).to_hash
+      end
   rescue Net::HTTPServerException, Chef::Exceptions::InvalidDataBagName
     node.run_state['letsencryptaws_creds'] = {}
   end
   node.run_state['letsencryptaws_creds'].fetch(key, nil)
 end
+
+def temp_credentials_path
```

Comment:
Just make this a node attribute please so that it is easily overridable.

Reply:
Just validating it all works as expected.

Reply:
The update works.